### PR TITLE
feat(console): add paths configuration on entrypoints screen

### PR DESCRIPTION
**Pull request by gcusnieux**
## Issue

https://gravitee.atlassian.net/browse/APIM-450

## Description

A small description of what you did in that PR.

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-kraegqjdox.chromatic.com)

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.z6.web.core.windows.net/apim-450-entrypoints-path/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

### CODE REVIEW

File: gravitee-apim-console-webui/src/management/api/creation/v4/components/gio-form-listeners/gio-form-listeners-context-path/gio-form-listeners-context-path.harness.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import { MatButtonHarness } from '@angular/material/button/testing';
+import { MatInputHarness } from '@angular/material/input/testing';
+
+import { HttpListenerPath } from '../../../../../../../entities/api-v4';
+
+export class GioFormListenersContextPathHarness extends ComponentHarness {
+  public static hostSelector = 'gio-form-listeners-context-path';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a `GioFormColorInputHarness` that meets
```
Comment (loriepisicchio) [inline gravitee-apim-console-webui/src/management/api/creation/v4/components/gio-form-listeners/gio-form-listeners-context-path/gio-form-listeners-context-path.harness.ts]:
```suggestion
   * Gets a `HarnessPredicate` that can be used to search for a `GioFormListenersContextPathHarness` that meets
```

File: gravitee-apim-console-webui/src/management/api/creation/v4/components/gio-form-listeners/gio-form-listeners-virtual-host/gio-form-listeners-virtual-host.harness.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+import { MatSlideToggleHarness } from '@angular/material/slide-toggle/testing';
+
+import { GioFormListenersContextPathHarness } from '../gio-form-listeners-context-path/gio-form-listeners-context-path.harness';
+import { HttpListenerPath } from '../../../../../../../entities/api-v4';
+
+export class GioFormListenersVirtualHostHarness extends GioFormListenersContextPathHarness {
+  public static hostSelector = 'gio-form-listeners-virtual-host';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a `GioFormColorInputHarness` that meets
```
Comment (loriepisicchio) [inline gravitee-apim-console-webui/src/management/api/creation/v4/components/gio-form-listeners/gio-form-listeners-virtual-host/gio-form-listeners-virtual-host.harness.ts]:
```suggestion
   * Gets a `HarnessPredicate` that can be used to search for a `GioFormListenersVirtualHostHarness` that meets
```

File: gravitee-apim-console-webui/src/management/api/creation/v4/components/gio-form-listeners/gio-form-listeners-context-path/gio-form-listeners-context-path.component.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+  constructor(private readonly fm: FocusMonitor, private readonly elRef: ElementRef) {
+    this.mainForm = new FormGroup({
+      listeners: this.listenerFormArray,
+    });
+  }
+
+  ngOnInit(): void {
+    // When user start to complete last listener add new empty one at the end
+    this.listenerFormArray?.valueChanges
+      .pipe(
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/creation/v4/components/gio-form-listeners/gio-form-listeners-context-path/gio-form-listeners-context-path.component.ts]: unsubscribe missing ;)

File: gravitee-apim-console-webui/src/management/api/creation/v4/components/gio-form-listeners/gio-form-listeners.common.scss [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  }
+
+  &__title {
+    font-size: 14px;
+    text-transform: none;
+    display: flex;
+    align-items: center;
+
+    &__icon {
+      color: #9698a2;
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/creation/v4/components/gio-form-listeners/gio-form-listeners.common.scss]: 🔗 to ui-p ?

File: gravitee-apim-console-webui/src/management/api/creation/v4/steps/step-2-entrypoints/step-2-entrypoints-2-config.component.html [hunk 5]
```diff
@@ -22,9 +22,27 @@
...
 
-  TODO Config context path or virtual host <br /><br />
-
   <form *ngIf="formGroup" [formGroup]="formGroup" (ngSubmit)="save()">
+    <div class="api-creation-v4__form-container">
+      <h3>Configure common entrypoints fields</h3>
+      <h4 class="api-creation-v4__form-container__subTitle">Entrypoints context</h4>
+      <gio-form-listeners-context-path
+        *ngIf="!this.enableVirtualHost"
+        [formControl]="getPathsFormControl()"
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/creation/v4/steps/step-2-entrypoints/step-2-entrypoints-2-config.component.html]: Try with `formControlName="paths"`

File: gravitee-apim-console-webui/src/components/gio-form-listeners/gio-form-listeners-context-path/gio-form-listeners-context-path.component.ts [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+            !this.listenerFormArray.controls[listeners.length - 1].invalid &&
+            !this.isEmpty(listeners[listeners.length - 1])
+          ) {
+            this.addEmptyListener();
+          }
+        }),
+      )
+      .subscribe();
+
+    this.fm.monitor(this.elRef.nativeElement, true).subscribe(() => {
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/components/gio-form-listeners/gio-form-listeners-context-path/gio-form-listeners-context-path.component.ts]:
```suggestion
    this.fm.monitor(this.elRef.nativeElement, true).pipe(takeUntil(this.unsubscribe$)).subscribe(() => {
```

**Review by ThibaudAV — APPROVED** [review]

**Review by loriepisicchio — APPROVED** [review]

**Pull request merged**